seattle-uat/civiform
# Write metadata for every question an applicant answered

### Main Updates and Files Changed
1. Add `updated_at` and `updated_in_program` metadata to questions themselves (i.e. they are scalars)
    1. _QuestionDefinition, NameQuestionDefinition, AddressQuestionDefinition, TextQuestionDefinition_
2. Change `ScalarType.INT` to `ScalarType.LONG` - we will represent all whole numbers as longs
    1. _ScalarType, ApplicantData (methods for int --> long)_
3. For each question path in updates for a block, write two pieces of metadata (updated time and program ID)
    1. _ApplicantServiceImpl + Test_
4. Remove `stageAndUpdateIfValid` method on `ApplicantService` that exposes internal `Update` details, and instead only expose the method which uses `ImmutableMap`
    1. _ApplicantService, ApplicantServiceImpl (reordered methods)_
5. Update `ApplicantData` methods to avoid writing the empty string
6. Update logic in `ApplicantQuestion` to check whether an applicant has answered a question, instead of checking for presence of a value
    1. _ApplicantQuestion + Test_

### Checklist

### Issue(s)
#519
